public/app/features/scenes/variables/variants/MultiValueVariable.ts

```diff
@@ -146,3 +167,38 @@ export abstract class MultiValueVariable<TState extends MultiValueVariableState
     test.setState(state);
   }
 }
+
+export class MultiValueUrlSyncHandler<TState extends MultiValueVariableState = MultiValueVariableState>
+  implements SceneObjectUrlSyncHandler<TState>
+{
+  public constructor(private _sceneObject: MultiValueVariable<TState>) {}
+
+  private getKey(): string {
+    return `var-${this._sceneObject.state.name}`;
+  }
+
+  public getKeys(): string[] {
+    return [this.getKey()];
+  }
+
+  public getUrlState(state: TState): SceneObjectUrlValues {
+    let urlValue: string | string[] | null = null;
+    let value = state.value;
+
+    if (Array.isArray(value)) {
+      urlValue = value.map(String);
+    } else {
+      urlValue = String(value);
+    }
+
+    return { [this.getKey()]: urlValue };
+  }
+
+  public updateFromUrl(values: SceneObjectUrlValues): void {
+    const urlValue = values[this.getKey()];
+
+    if (urlValue != null) {
+      this._sceneObject.changeValueTo(urlValue);
+    }
+  }
+}
```

Comment:
From one side, I like to have this as a different instance of implementing the interface. It keeps the logic encapsulated. 

But, from the other side, maybe it would be easier to just make those functions part of the SceneObject, so you can access scene props and state using `this`. In the end, we need the sceneObject instance to interact with the scene object, so it makes sense to have those functions as part of the scene object to access easily the instance and state of it. 

Probably this is the feedback I could have given in the other PR, but when I see it here, more atomically applied, I see clearly that it might be confusing and hard to understand if the instance of the URL sync is out of the object.

Reply:
@ivanortegaalba true, but I like that it's isolated like his as it makes it a bit easier to reason about code and its responsibility.  These functions will be a bit lost when added to the main class

Reply:
I actually prefer the separation. I think the URL sync is like a mixin in the scene architecture, while the SceneObject interface is primarily about the scene object lifecycle. This clearly separates the responsibility and keeps th scene object interface small 👍